deltacat/tests/compute/test_compact_partition_rebase.py

```diff
@@ -0,0 +1,276 @@
+    create_placement_group_param: bool,
+    records_per_compacted_file_param: int,
+    hash_bucket_count_param: int,
+    drop_duplicates_param: bool,
+    read_kwargs_provider_param: Any,
+    rebase_expected_compact_partition_result: pa.Table,
+    skip_enabled_compact_partition_drivers: List[CompactorVersion],
+    compact_partition_func: Callable,
+    benchmark: BenchmarkFixture,
+):
+    import deltacat.tests.local_deltacat_storage as ds
+
+    ds_mock_kwargs = local_deltacat_storage_kwargs
+    """
+    This test tests the scenario where source partition locator == destination partition locator,
+    but rebase source partition locator is different.
+    This scenario could occur when hash bucket count changes.
+    """
+    partition_keys = partition_keys_param
+    (
+        source_table_stream,
+        _,
+        rebased_table_stream,
+    ) = create_src_w_deltas_destination_rebase_w_deltas_strategy(
+        primary_keys,
+        sort_keys,
+        partition_keys,
+        input_deltas_param,
+        input_deltas_delta_type,
+        partition_values_param,
+        ds_mock_kwargs,
+    )
+    source_partition: Partition = ds.get_partition(
+        source_table_stream.locator,
+        partition_values_param,
+        **ds_mock_kwargs,
+    )
+    rebased_partition: Partition = ds.get_partition(
+        rebased_table_stream.locator,
+        partition_values_param,
+        **ds_mock_kwargs,
+    )
+    num_workers, worker_instance_cpu = DEFAULT_NUM_WORKERS, DEFAULT_WORKER_INSTANCE_CPUS
+    total_cpus = num_workers * worker_instance_cpu
+    pgm = None
+    create_placement_group_param = False
+    if create_placement_group_param:
+        pgm = PlacementGroupManager(
+            1, total_cpus, worker_instance_cpu, memory_per_bundle=4000000
+        ).pgs[0]
+    compact_partition_params = CompactPartitionParams.of(
+        {
+            "compaction_artifact_s3_bucket": TEST_S3_RCF_BUCKET_NAME,
+            "compacted_file_content_type": ContentType.PARQUET,
+            "dd_max_parallelism_ratio": 1.0,
+            "deltacat_storage": ds,
+            "deltacat_storage_kwargs": ds_mock_kwargs,
+            "destination_partition_locator": rebased_partition.locator,
+            "hash_bucket_count": hash_bucket_count_param,
+            "last_stream_position_to_compact": source_partition.stream_position,
+            "list_deltas_kwargs": {**ds_mock_kwargs, **{"equivalent_table_types": []}},
+            "object_store": RayPlasmaObjectStore(),
+            "pg_config": pgm,
+            "primary_keys": primary_keys,
+            "read_kwargs_provider": read_kwargs_provider_param,
+            "rebase_source_partition_locator": source_partition.locator,
+            "records_per_compacted_file": records_per_compacted_file_param,
+            "s3_client_kwargs": {},
+            "source_partition_locator": rebased_partition.locator,
+            "sort_keys": sort_keys if sort_keys else None,
```

Comment:
I think you can use `sort_keys or None` here

Reply:
Would that still work if sort_keys is an empty data structure?